File: 1. Multiples of 3 or 5/solution.py
def sum_of_multiples(limit, divisor):
    n = (limit - 1) // divisor
    return divisor * n * (n + 1) // 2

def sum_multiples_3_or_5(limit):
    sum_multiples_3 = sum_of_multiples(limit, 3)
    sum_multiples_5 = sum_of_multiples(limit, 5)
    sum_multiples_15 = sum_of_multiples(limit, 15)
    return sum_multiples_3 + sum_multiples_5 - sum_multiples_15

result = sum_multiples_3_or_5(1000)
print(result)

File: 1. Multiples of 3 or 5/solution.java
public class solution {
    public static void main(String[] args) {
        int limit = 1000;
        int result = sumMultiples(limit);
        System.out.println(result);
    }

    public static int sumMultiples(int limit) {
        return sumDivisibleBy(3, limit) + sumDivisibleBy(5, limit) - sumDivisibleBy(15, limit);
    }

    public static int sumDivisibleBy(int divisor, int limit) {
        int n = (limit - 1) / divisor;
        return divisor * (n * (n + 1)) / 2;
    }
}